amplify/backend/function/neptuneDelete/src/index.py

```diff
@@ -0,0 +1,45 @@
+import json
+import os
+import gremlin_python
+from gremlin_python.driver import client, serializer
+
+def lambda_handler(event, context):
+    endpoint = os.environ['NEPTUNE_ENDPOINT']
+    port = os.environ['NEPTUNE_PORT']
+
+    try:
+        method = event["method"]
+    except:
+        return {
+        "statusCode": 400,
+        "body": "Bad Request: \"method\" parameter required."
+        }
+
+    # Connect to Neptune
+    db = client.Client(f"wss://{endpoint}:{port}/gremlin", "g")
+
+    # DB rest functions
+    def db_delete_user(username):
+        query = f"g.V().hasLabel('user').has('username', '{username}').drop()"
+        try:
+            db.submit(query)
+            return "Query executed successfully"
+        except:
+            return "Query failed"
+
+    result = ""
+
+    if method == "DELETE":
+        username = event["user"]
+        result = db_delete_user(username)
+    else:
+        return {
+            "statusCode": 400,
+            "body": "Bad Request: \"method\" parameter must be \"DELETE\"."
+        }
```

Comment:
Shouldn't have to check the HTTP method at the Lambda function level as its already validated at the API Gateway level.

That is, the only way to get to this Lambda function is if you send a `DELETE` method request to the correct resource. In your case, the `DELETE /api` you defined in API gateway.